logstash-core/lib/logstash/config/source/multi_local.rb

```diff
@@ -37,13 +37,13 @@ def pipeline_configs
       end
       detect_duplicate_pipelines(pipelines_settings)
       pipeline_configs = pipelines_settings.map do |pipeline_settings|
-        @settings = pipeline_settings
+        update_settings(pipeline_settings)
         # this relies on instance variable @settings and the parent class' pipeline_configs
         # method. The alternative is to refactor most of the Local source methods to accept
         # a settings object instead of relying on @settings.
         local_pipeline_configs # create a PipelineConfig object based on @settings
       end.flatten
-      @settings = @original_settings
+      settings = @original_settings
```

Comment:
:warning:  seems to be doing something else than the original

Reply:
fixed